core/store/src/flat/store_helper.rs

```diff
@@ -97,9 +96,32 @@ pub fn remove_flat_head(store_update: &mut StoreUpdate, shard_id: ShardId) {
     store_update.delete(FlatStateColumn::Misc.to_db_col(), &flat_head_key(shard_id));
 }
 
-pub(crate) fn get_ref(store: &Store, key: &[u8]) -> Result<Option<ValueRef>, FlatStorageError> {
+fn encode_flat_state_db_key(shard_uid: ShardUId, key: &[u8]) -> Vec<u8> {
+    let mut buffer = vec![];
+    buffer.extend_from_slice(&shard_uid.to_bytes());
+    buffer.extend_from_slice(key);
+    buffer
+}
+
+fn decode_flat_state_db_key(key: &Box<[u8]>) -> Result<(ShardUId, Vec<u8>), StorageError> {
+    if key.len() < 8 {
+        return Err(StorageError::StorageInconsistentState(format!(
+            "Found key in flat storage with length < 8: {key:?}"
+        )));
+    }
+    let shard_uid = key[0..8].try_into().unwrap();
+    let trie_key = key[8..].to_vec();
```

Comment:
nit: use [`split_at`](https://doc.rust-lang.org/std/primitive.slice.html#method.split_at) and avoid `unwrap`:
```
let (shard_uid_bytes, trie_key) = key.split_at(8);
let shard_uid = shard_uid_bytes.try_into().map_err(|_| StorageError::StorageInconsistentState(..))?;
Ok((shard_uid, trie_key.to_vec())
```

Reply:
Nice!